core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -192,12 +194,24 @@ void logResponseFirstBytesTransferred() {
 
         @Override
         public void run() {
-            final ResponseTimeoutException cause = ResponseTimeoutException.get();
-            delegate.close(cause);
-            logBuilder.endResponse(cause);
+            final Runnable responseTimeoutHandler;
+            if (logBuilder instanceof DefaultRequestLog) {
+                responseTimeoutHandler =
+                        ((ClientRequestContext) ((RequestLog) logBuilder).context()).responseTimeoutHandler();
```

Comment:
How about just passing `ctx` instead of `logBuilder` when creating `HttpResponseWrapper` so we don't need to do this type casting?

Reply:
Originally, I thought of passing `Supplier<Runnable> responseTimeoutHandlerSupplier` or `ctx` too. But I don't have a strong convince whether it is good for not. 

The disadvantage of this is that we should cast `logBuilder`. However `RequestContext` and `ClientRequestContext` is mutable, so I wanted to stay `ctx` in `HttpSessionHandler` and hide it in `HttpResponseDecoder`. Let me know if you think it would be better to add `ctx` as a parameter.

Reply:
> However RequestContext and ClientRequestContext is mutable

Is there any difference if we pass the `ctx` instread of using `logBuilder` to retrieve the `ctx` in the point of view of mutability?

Reply:
There is no difference in mutability. IMHO, the main difference is the scope of `ctx`. We can directly access `ctx` from `logBuilder`. In this case, `ctx` is only visible in this `if` block. :-) As you know, if we pass `ctx` with the constructor, it could be accessed everywhere in this class. 

Either is fine. But this idea was suggested by @trustin. I wanted to hear his opinion 👂

Reply:
+1 for just passing `ctx` if possible.

Reply:
Fixed it https://github.com/line/armeria/pull/2248/commits/f944b32506c1d743a12cee3a01f52c7abe98a793